11.coreboot.u-root.systemboot/README.md

```diff
@@ -43,7 +43,116 @@ For example, you may want to include static builds of `kexec` or `flashrom`, tha
 
 ## Building a suitable Linux kernel
 
-TODO
+TODO add sample `.config`.
+
+You need a relatively recent kernel. Ideally a kernel 4.16, to have support for VPD variables, but a 4.11 can do the job too, if you don't care about boot entries and want "brute-force" booting only.
+
+We will build a kernel with the following properties:
+
+* small enough to fit most flash chips
+* that can run Go programs (mainly futex and epoll support)
+* with the relevant storage and network drivers
+* with kexec support, so it can boot a new kernel
+* with kexec signature verification disabled (optional)
+* with devtmpfs enabled, since we don't use udev
+* XZ support to decompress the embedded initramfs
+* VPD (Vital Product Data) (optional)
+* TPM support (optional)
+* embed the u-root initramfs
+* and last but not least, "linuxboot" as default host name :)
+
+
+### Download kernel sources
+
+You can either download a tarball from kernel.org, or get it via git and use a
+version tag. We recommend at least a kernel 4.16, in order to have VPD variables
+support.
+
+```
+# download the kernel tarball. Replace 4.19.6` with whatever kernel version you want
+wget https://cdn.kernel.org/pub/linux/kernel/v4.x/linux-4.19.6.tar.xz
+tar xvJf linux-4.19.6.tar.xz
+cd linux-4.19.6
+make tinyconfig
+```
+
+Some more information about tiny configs can be found at https://tiny.wiki.kernel.org (last checked 2018-12-01).
+
+### Requirements for Go 1.11
+
+`Go` requires a few kernel features to work properly. At the time of writing,
+you need to enable `CONFIG_FUTEX` in your kernel config.
```

Comment:
you also require CONFIG_EPOLL last time I checked. This is new post about 1.8 or something.

Reply:
I removed it since it's not documented anymore on the Golang wiki - I believe it's not a requirement anymore. I can reintroduce it though, and mention that it *may* be required

Reply:
I now mentioned that on older Go versions CONFIG_EPOLL may be required